src/loaders/jobs/taskfarmer/taskfarmer_task_mgr.py

```diff
@@ -0,0 +1,197 @@
+
+        task_info_file = self._get_task_info_file()
+        df = pd.read_json(task_info_file, lines=True)
+
+        tasks_df = df[(df["kbase_collection"] == self.kbase_collection) &
+                      (df["load_ver"] == self.load_ver) &
+                      (df["tool"] == self.tool)]
+
+        tasks_df.sort_values(by="job_submit_time", ascending=False, inplace=True)
+
+        return tasks_df
+
+    def _create_job_dir(self, destroy_old_job_dir=False):
+        """
+        Create the job directory. If destroy_old_job_dir is True, recreate the job directory.
+        """
+
+        if os.path.exists(self.job_dir) and destroy_old_job_dir:
+            print(f'removing job dir {self.job_dir}')
+            shutil.rmtree(self.job_dir, ignore_errors=True)
+
+        os.makedirs(self.job_dir, exist_ok=True)
+
+        return self.job_dir
+
+    def _get_job_status_from_nersc(self, job_id):
+        """
+        Get the job status from NERSC using squeue command.
+
+        A 'done.tasks.txt.tfin' file is created in the job directory when the job is finished successfully.
+
+        When job is queued, squeue command will return code 0 and out put will be like (with no job info just the header):
+        JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+
+
+        When job is running, squeue command will return code 0 and out put will be like:
+        JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+        5987779 regular_m submit_t      tgu  R       1:20      2 nid[005689-005690]
+
+        When job is finished/failed:
+            * job status is cached for a period of time (about 10 mins). squeue command will return code 0 and
+            out put will be like below (with no job info just the header):
+            JOBID PARTITION     NAME     USER ST       TIME  NODES NODELIST(REASON)
+
+            * after the period of time, squeue command will return error code 1 and error message:
+            slurm_load_jobs error: Invalid job id specified
+
+        Matrix of job status:
+        Done file  Exit Code   Job listed    Status
+        Y          Any         Any           COMPLETED
+        N          0           N             PENDING or FAILED (if failed, exit code will flip to -1 within ~10m)
+        N          0           Y             RUNNING
+        N          -1          N/A           FAILED
+
+        NOTE: If job fails and squeue command hasn't flipped the exit code to -1, this function will return 'PENDING'
+        as the job status. After about 10 mins, the exit code will be flipped to -1 and this function will return
+        'FAILED' status as the job status.
+        """
+
+        # Once all tasks are completed, NERSC generates this file.
+        # (ref: https://docs.nersc.gov/jobs/workflow/taskfarmer/#output)
+        done_file = os.path.join(self.job_dir, NERSC_SLURM_DONE_FILE)
+        if os.path.isfile(done_file):
+            return JobStatus.COMPLETED
```

Comment:
I'm confused here as well. What if you start a job, it finishes, and then you start a new job (say with a different source directory) with a new job_id. This function will still report the job as done, won't it?

Reply:
you can only start a new job by calling `task_generator.py` script which initializes TaskManager. At this point, task returned by the TaskManager will be the latest job created.

Reply:
when `_submit_job` is called, all `.tfin` files will be removed first. that including the done file. If that's what you are asking.

Reply:
Maybe _submit_job should be a task manager method so that all the done file handling is encapsulated in the task manager?

Reply:
👍

Reply:
:+1: